src/TLuaInterpreter.cpp

```diff
@@ -3547,6 +3565,47 @@ int TLuaInterpreter::createMapper(lua_State* L)
     return 1;
 }
 
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#createVideoPlayer
+int TLuaInterpreter::createVideoPlayer(lua_State* L)
+{
+    int n = lua_gettop(L);
+    QString windowName = "";
+    int counter = 1;
+
+    if (n > 4) {
+        if (lua_type(L, 1) != LUA_TSTRING) {
+            lua_pushfstring(L, "createVideoPlayer: bad argument #1 type (parent window name as string expected, got %s!)", luaL_typename(L, 1));
+            return lua_error(L);
+        }
+
+        windowName = lua_tostring(L, 1);
+
+        counter++;
+
+        if (isMain(windowName)) {
+            // createVideoPlayer only accepts the empty name as the main window
+            windowName.clear();
+        }
+    }
+
+    int x = getVerifiedInt(L, __func__, counter, "video player x-coordinate");
+    counter++;
+    int y = getVerifiedInt(L, __func__, counter, "video player y-coordinate");
+    counter++;
+    int width = getVerifiedInt(L, __func__, counter, "video player width");
+    counter++;
+    int height = getVerifiedInt(L, __func__, counter, "video player height");
```

Comment:
More efficient coding:
```suggestion
    int x = getVerifiedInt(L, __func__, counter, "video player x-coordinate");
    int y = getVerifiedInt(L, __func__, ++counter, "video player y-coordinate");
    int width = getVerifiedInt(L, __func__, ++counter, "video player width");
    int height = getVerifiedInt(L, __func__, ++counter, "video player height");
```